src/applications/vaos/referral-appointments/ScheduleReferral.jsx

```diff
@@ -1,65 +1,47 @@
-import React, { useEffect, useState } from 'react';
+import React, { useEffect } from 'react';
 import { useParams, useLocation } from 'react-router-dom';
 import { useDispatch } from 'react-redux';
 import { format } from 'date-fns';
 import { VaAlert } from '@department-of-veterans-affairs/component-library/dist/react-bindings';
 import FormLayout from '../new-appointment/components/FormLayout';
 import ReferralAppLink from './components/ReferralAppLink';
-import { getPatientReferralById } from '../services/referral';
 import { setFormCurrentPage } from './redux/actions';
-import { referral } from './temp-data/referral';
+import { FETCH_STATUS } from '../utils/constants';
+import { scrollAndFocus } from '../utils/scrollAndFocus';
+import { useGetReferralFromId } from './hooks/useGetReferralFromId';
 
 export default function ScheduleReferral() {
-  const [patientData, setPatientData] = useState({});
-  const [loading, setLoading] = useState(true);
-  const [error, setError] = useState(false);
   const location = useLocation();
   const dispatch = useDispatch();
   const params = useParams();
-
+  const { id } = params;
+  const { currentReferral, referralFetchStatus } = useGetReferralFromId(id);
   useEffect(
     () => {
       dispatch(setFormCurrentPage('scheduleReferral'));
     },
     [location, dispatch],
   );
-
   useEffect(
     () => {
-      const fetchDetails = async () => {
-        let details = {};
-        setError(false);
-        try {
-          // get the id from the url where url is /referral-review/:id
-          const { id } = params;
-          details = await getPatientReferralById(id);
-        } catch (networkError) {
-          setError(true);
-        }
-        setPatientData(details);
-        setLoading(false);
-      };
-
-      // @TODO: This will eventually check for data in selected from redux
-      if (!referral) {
-        fetchDetails();
-      } else {
-        setPatientData(referral);
-        setLoading(false);
+      if (referralFetchStatus === FETCH_STATUS.succeeded) {
+        scrollAndFocus('h1');
+      } else if (referralFetchStatus === FETCH_STATUS.failed) {
+        scrollAndFocus('h2');
       }
     },
-    [params],
+    [referralFetchStatus],
   );
 
-  if (loading) {
+  if (!currentReferral || referralFetchStatus === FETCH_STATUS.loading) {
```

Comment:
Because of this check all later `currentReferral?`s are probably unnecessary

Reply:
makes sense, I'll fix that up.